evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -42,19 +47,34 @@ def validate(self, X, y):
-            messages["errors"].append(DataCheckError("{} row(s) ({}%) of target values are null".format(null_rows.sum(), null_rows.mean() * 100), self.name).to_dict())
+            num_null_rows = null_rows.sum()
+            pct_null_rows = null_rows.mean() * 100
+            messages["errors"].append(DataCheckError(message="{} row(s) ({}%) of target values are null".format(num_null_rows, pct_null_rows),
+                                                     data_check_name=self.name,
+                                                     message_code=DataCheckMessageCode.TARGET_HAS_NULL,
+                                                     details={"num_null_rows": num_null_rows, "pct_null_rows": pct_null_rows}).to_dict())
         valid_target_types = [dtype for dtype in numeric_and_boolean_dtypes + categorical_dtypes]
         if y.dtype.name not in valid_target_types:
-            messages["errors"].append(DataCheckError("Target is unsupported {} type. Valid target types include: {}".format(y.dtype, ", ".join(valid_target_types)), self.name).to_dict())
+
+            messages["errors"].append(DataCheckError(message="Target is unsupported {} type. Valid target types include: {}".format(y.dtype, ", ".join(valid_target_types)),
+                                                     data_check_name=self.name,
+                                                     message_code=DataCheckMessageCode.TARGET_UNSUPPORTED_TYPE,
+                                                     details={"unsupported_type": y.dtype}).to_dict())
 
         value_counts = y.value_counts()
+        unique_values = value_counts.index.tolist()
 
         if self.problem_type == ProblemTypes.BINARY and len(value_counts) != 2:
-            messages["errors"].append(DataCheckError("Target does not have two unique values which is not supported for binary classification", self.name).to_dict())
+            messages["errors"].append(DataCheckError(message="Target does not have two unique values which is not supported for binary classification",
+                                                     data_check_name=self.name,
+                                                     message_code=DataCheckMessageCode.TARGET_BINARY_NOT_TWO_UNIQUE_VALUES,
+                                                     details={"target_values": unique_values}).to_dict())
 
         if len(value_counts) == 2 and y.dtype in numeric_and_boolean_dtypes:
-            unique_values = value_counts.index.tolist()
             if set(unique_values) != set([0, 1]):
-                messages["errors"].append(DataCheckError("Numerical binary classification target classes must be [0, 1], got [{}] instead".format(", ".join([str(val) for val in unique_values])), self.name).to_dict())
+                messages["warnings"].append(DataCheckWarning(message="Numerical binary classification target classes must be [0, 1], got [{}] instead".format(", ".join([str(val) for val in unique_values])),
```

Comment:
@dsherry Right, but that's why we have `problem_type` passed in for the initialization of `InvalidTargetDataCheck`, right? Or are you concerned if the user accidentally passes in the wrong problem type?

Reply:
Yep, that's what I was thinking, if someone accidentally ran binary classification with a regression target. One quick way around this would be to show the number of uniques and the counts of the 100 most frequent uniques instead of the counts of all the uniques!

No need to add this to this PR heh. But this would be great to file as a performance enhancement.

Reply:
Ah, got it! Filed #1460 :)